simd/src/Kokkos_SIMD_NEON.hpp

```diff
@@ -965,6 +965,13 @@ class const_where_expression<
     if (m_mask[0]) mem[0] = m_value[0];
     if (m_mask[1]) mem[1] = m_value[1];
   }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+  void scatter_to(
+      std::int32_t* mem,
+      simd<std::int32_t, simd_abi::neon_fixed_size<2>> const& index) const {
+    m_value[0] = (m_mask[0]) ? mem[0] : 0;
```

Comment:
Wait I thought `scatter_to` would write to `mem` not read from it.
What am I missing?